src/common/snippets/include/snippets/op/scalar.hpp

```diff
@@ -16,29 +16,30 @@ namespace op {
  * @brief Generated by Canonicalization for a scalar constant Shape() == {1}
  * @ingroup snippets
  */
-class Scalar  : public ngraph::op::Constant {
+class Scalar  : public ov::op::v0::Constant {
 public:
-    OPENVINO_OP("Scalar", "SnippetsOpset", ngraph::op::Constant);
+    OPENVINO_OP("Scalar", "SnippetsOpset", ov::op::v0::Constant);
 
-    Scalar() = default;
-    Scalar(const std::shared_ptr<runtime::Tensor>& tensor) : Constant(tensor) {}
-    template <typename T>
-    Scalar(const element::Type& type, Shape shape, const std::vector<T>& values) : Constant(type, shape, values) {}
-    Scalar(const element::Type& type, const Shape& shape) : Constant(type, shape) {}
     template <class T, class = typename std::enable_if<std::is_fundamental<T>::value>::type>
-    Scalar(const element::Type& type, Shape shape, T value) : Constant(type, shape, value) {}
-    Scalar(const element::Type& type, Shape shape, const std::vector<std::string>& values) : Constant(type, shape, values) {}
-    Scalar(const element::Type& type, const Shape& shape, const void* data) : Constant(type, shape, data) {}
-
-    Scalar(const Constant& other) : Constant(other) {}
-    Scalar(const Scalar& other) : Constant(other) {}
+    Scalar(const element::Type& type, Shape shape, T value) : Constant(type, shape, value) {validateScalarShape();}
+    explicit Scalar(const Constant& other) : Constant(other) {validateScalarShape();}
+    Scalar(const Scalar& other) : Constant(other) {validateScalarShape();}
     Scalar& operator=(const Scalar&) = delete;
-    ~Scalar() override {}
 
     std::shared_ptr<Node> clone_with_new_inputs(const OutputVector& new_args) const override {
         check_new_args_count(this, new_args);
         return std::make_shared<Scalar>(*this);
     }
+
+private:
+    // Scalar currently supports only one-element constants, this could be changed in the future
+    void validateScalarShape() {
```

Comment:
Yes, we can. I agree that it's ok to skip checking scalar shape, but only until it's called exclusively from convert_to_snippets_dialect, which is followed by validate_and_infer_types (which is the case now).

Otherwise it might be difficult to trace where the invalid scalar was created. It's probably a good idea to insert constructor_validate_and_infer_types() in Scalar constructors to handle such cases. What do you think?

Reply:
Fully agree. Lets add `constructor_validate_and_infer_types()` as well

Reply:
Ok, added.